src/providers/address/address.ts

```diff
@@ -24,80 +24,45 @@ export class AddressProvider {
     };
   }
 
-  public getCoin(address: string) {
-    address = address.replace(/^(bitcoincash:|bchtest:|bitcoin:)/i, '');
+  public getCoin(address: string): string {
+    let extractedAddress = this.extractAddress(address);
```

Comment:
it is difficult to understand the function like it is written:


in `extractAddress(address)` is not obvious what is happending. Why extracting an address from something that is already `address`?

Reply:
maybe `removePrefix`?

Reply:
That function also removes suffixes from uris, for example: 
`'bitcoin:1CVuVALD6Zo7ms24n3iUXv162kvUzsHr69?amount=0.00090000'`

The function returns: ```1CVuVALD6Zo7ms24n3iUXv162kvUzsHr69```
Maybe I can use ```cleanAddress``` ?